Android Source/mobile/src/main/java/net/steppschuh/sensordatalogger/DataRecorder.java

```diff
@@ -0,0 +1,140 @@
+package net.steppschuh.sensordatalogger;
+
+import android.os.Environment;
+import android.util.Log;
+
+import net.steppschuh.datalogger.data.DataBatch;
+import net.steppschuh.datalogger.data.DataPoint;
+import net.steppschuh.datalogger.sensor.DeviceSensor;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class DataRecorder {
+
+    private static final String TAG = DataRecorder.class.getSimpleName();
+    private static final String CSV_HEADER = "timestamp,accuracy,x,y,z";
```

Comment:
Not all sensors produce three-dimensional data, should be based on available data length